### PR TITLE
Fix topk_scores comparison in test_moe_gate_prefill2d.py to prevent significant PCC drop on all models (especially Kimi K3)

### PR DESCRIPTION
**Pull request by pmilojevicTT**
### Ticket
https://github.com/tenstorrent/tt-metal/issues/52569

### Problem

Current problem is that we are comparing `topk_scores` tensor on host and device by classic PCC comparison. That causes PCC drop, because `topk_scores` is actual result of `sigmoid(logits)`, but `topk_indices` and `topk_scores` have elements order ordered in descending order based on result of `sigmoid(logits)+bias`, which can cause that best score of `sigmoid(logits)` on host and device doesn't have to be from exact same expert, but we are comparing it as it's from exactly same expert. Order of experts for each token (which expert is best, which is second best and so on) isn't important in E2E prefill scenario, only which `topK` experts are chosen for each token is important and we already compare that good using `recall` comparison instead of `PCC` for `topk_indices` tensor on host and device.

### What's changed

- sort `topk_scores` tensor in descending order, because exact order of experts for each chip isn't important so this is best way to compare best expert on host with best expert on device and other places in order too
- increase threshold for `GATE_SCORES_PCC_DEVICE` for models.

### CI

- [x] [(BH) e2e
tests](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-e2e-tests.yaml?query=branch:pmilojevic/52569)
- [x] [(Galaxy) Blaze Models Prefill tests](https://github.com/tenstorrent/tt-metal/actions/workflows/blaze-models-prefill-tests.yaml?query=branch:pmilojevic/52569)

### CI Status
_Auto-generated on every push. Badges update live. Click a badge to filter runs by this branch._

- [](https://github.com/tenstorrent/tt-metal/actions/workflows/sanity-tests.yaml?query=branch:pmilojevic/52569)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/runtime-sanity-tests.yaml?query=branch:pmilojevic/52569)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/tt-metal-l2-nightly.yaml?query=branch:pmilojevic/52569)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/all-model-tests.yaml?query=branch:pmilojevic/52569)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/pipeline-select.yaml?query=branch:pmilojevic/52569)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/pipeline-select-t3k.yaml?query=branch:pmilojevic/52569)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/pipeline-select-galaxy.yaml?query=branch:pmilojevic/52569)

### CODE REVIEW

File: models/demos/deepseek_v3_d_p/reference/kimi_k3_config.py [hunk 1]
```diff
@@ -55,10 +55,11 @@ class KimiK3Config:
...
 
-    # Gate-test device-mode scores bar, relaxing the shared 0.93; see #52569. 896 experts under sigmoid
-    # near-tie the 16th and 17th scores often enough that device precision swaps a pick, and the
-    # spread across Blackhole Galaxies (0.886 - 0.952) straddles the shared bar.
-    GATE_SCORES_PCC_DEVICE = 0.87
+    # Gate-test device-mode scores bar, tightening the shared 0.93; see #52569. 896 experts under
+    # sigmoid near-tie the 16th and 17th scores often enough that device precision swaps a pick, and
+    # the spread across Blackhole Galaxies is 0.886 - 0.952, so this bar sits inside that spread
+    # rather than below it.
+    GATE_SCORES_PCC_DEVICE = 0.94
```
[the remark this answers is not in the document]
Comment (pmilojevicTT) [reply models/demos/deepseek_v3_d_p/reference/kimi_k3_config.py]: This passes in CI

**Review by kgrujcic-tt — APPROVED** [review]
Ensure the CIs are passing.
